### PR TITLE
Rest/implementation

### PR DESCRIPTION
**Pull request by ngu-oli**

### CODE REVIEW

File: src/rest/screenplay/actions/Delete.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        this.headers = headers;
+        return this;
+    }
+
+    /**
+     * Save the response to this HTTP request. After the request is executed, the response to the request will be written into the function argument.
+     *
+     * @param response the response to the request will be written into this function argument
+     */
+    public andSaveResponseTo(response: Response): Delete {
```
Comment (kbrueckner) [inline src/rest/screenplay/actions/Delete.ts:63]: naming is weird here, I suppose send would be a better fit.

File: src/rest/types.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/**
+ * Methods for http requests
+ */
+export enum REQUEST_METHOD {
```
Comment (kbrueckner) [inline src/rest/types.ts:4]: This should be in an enum or constants file

File: src/rest/types.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * InternalRequestMethod type for internal handling of the universal request method
+ */
+export type RequestMethod = REQUEST_METHOD;
+
+/**
+ * HTTP Header type
+ */
+export type Header = {
```
Comment (kbrueckner) [inline src/rest/types.ts:21]: rename to headers

File: src/rest/types.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    [key: string]: any
+};
+
+/**
+ * Response type which is returned from any request
+ */
+export type Response = {
+    body: any;
```
Comment (kbrueckner) [inline src/rest/types.ts:29]: should be more specific: Object (for json), string (for text) or undefined/null

**Comment by kbrueckner** [comment]
Also README updates missing :)

**Review by munichbughunter — APPROVED** [review]

**Pull request merged**